src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> optionalUserByEmail = userService.findByEmail(email);
+        if (optionalUserByEmail.isEmpty()
+                || optionalUserByEmail.get().getPassword()
+                .equals(HashUtil.hashPassword(password,
+                        optionalUserByEmail.get().getSalt()))) {
```

Comment:
let's create a separate method: `checkPassword(User user, String password)` and use it in `if` block